core/src/avm1/value.rs

```diff
@@ -126,68 +126,25 @@ impl<'gc> Value<'gc> {
     /// expected that their `toString`/`valueOf` handlers have already had a
     /// chance to unbox the primitive contained within.
     pub fn is_primitive(&self) -> bool {
-        !matches!(self, Value::Object(_))
-    }
-
-    /// ECMA-262 2nd edition s. 9.3 ToNumber (after calling `to_primitive_num`)
-    ///
-    /// Flash diverges from spec in a number of ways. These ways are, as far as
-    /// we are aware, version-gated:
-    ///
-    /// * In SWF6 and lower, `undefined` is coerced to `0.0` (like `false`)
-    /// rather than `NaN` as required by spec.
-    /// * In SWF5 and lower, hexadecimal is unsupported.
-    /// * In SWF4 and lower, `0.0` is returned rather than `NaN` if a string cannot
-    /// be converted to a number.
-    fn primitive_as_number(&self, activation: &mut Activation<'_, 'gc, '_>) -> f64 {
-        match self {
-            Value::Undefined if activation.swf_version() < 7 => 0.0,
-            Value::Null if activation.swf_version() < 7 => 0.0,
-            Value::Object(_) if activation.swf_version() < 5 => 0.0,
-            Value::Undefined => f64::NAN,
-            Value::Null => f64::NAN,
-            Value::Bool(false) => 0.0,
-            Value::Bool(true) => 1.0,
-            Value::Number(v) => *v,
-            Value::Object(_) => f64::NAN,
-            Value::String(v) => string_to_f64(v, activation.swf_version()),
-        }
+        !matches!(self, Value::Object(object) if object.as_display_object().is_none())
```

Comment:
this has nothing to do with `is_primitive` though?

Reply:
Actually, I think it's a more precise separation between primitives and objects, as MovieClips are treated as primitives in most cases.

Once we'll have separate `Value::Object` and `Value::MovieClip` variants, this will be removed.

Reply:
> Once we'll have separate Value::Object and Value::MovieClip

I'm not sure that's a given, though? I'm not saying it's not _a_ solution, but I don't think this was ever directly proposed when we discussed MC references on discord.

Reply:
That said, it _does_ grow on me as an idea :) Was just really confused when you presented it as foregone conclusion, as I don't recall anyone mentioning this before.